src/render_engine/vulkan/vulkan_render_engine.cpp

```diff
@@ -1208,6 +1211,56 @@ namespace nova::renderer::rhi {
         NOVA_CHECK_RESULT(vkCreateDebugUtilsMessengerEXT(instance, &debug_create_info, nullptr, &debug_callback));
     }
 
+    void VulkanRenderEngine::save_device_info() {
+        switch(gpu.props.vendorID) {
+            case AMD_PCI_VENDOR_ID:
+                info.architecture = DeviceArchitecture::Amd;
+                break;
+
+            case INTEL_PCI_VENDOR_ID:
+                info.architecture = DeviceArchitecture::Intel;
+                break;
+
+            case NVIDIA_PCI_VENDOR_ID:
+                info.architecture = DeviceArchitecture::Nvidia;
+                break;
+
+            default:
+                info.architecture = DeviceArchitecture::Unknown;
+        }
+
+        vk_info.max_uniform_buffer_size = gpu.props.limits.maxUniformBufferRange;
+        info.max_texture_size = gpu.props.limits.maxImageDimension2D;
+
+        // TODO: Something smarter when Intel releases discreet GPUS
+        // TODO: Handle integrated AMD GPUs
+        info.is_uma = info.architecture == DeviceArchitecture::Intel;
+
+        uint32_t extension_count;
+        vkEnumerateDeviceExtensionProperties(gpu.phys_device, nullptr, &extension_count, nullptr);
+        std::vector<VkExtensionProperties> available_extensions(extension_count);
+        vkEnumerateDeviceExtensionProperties(gpu.phys_device, nullptr, &extension_count, available_extensions.data());
+
+        auto extension_name_matcher = [](const char* ext_name, const VkExtensionProperties& ext_props) -> bool {
+            return std::strcmp(ext_name, ext_props.extensionName) == 0;
+        };
+
+        // TODO: use std::bind_front instead of std::bind when C++20 drops
+        using namespace std::placeholders;
+
+        // TODO: Update as more GPUs support hardware raytracing
+        const auto rt_ext_itr = std::find_if(available_extensions.begin(),
+                                             available_extensions.end(),
+                                             std::bind(extension_name_matcher, VK_NV_RAY_TRACING_EXTENSION_NAME, _1));
```

Comment:
bind considered harmful

Reply:
C++20 hasn't given us `std::bind_front` yet so idk. What should I use instead?

Reply:
A lambda? 

```cpp
[](auto x) { extension_name_matcher(VK_NV_RAY_TRACING_EXTENSION_NAME, x)
```
or something similar

Reply:
I hate it but I hate everything so it's fine